website/docs/migration/1.0.0_to_2.0.0.mdx

```diff
@@ -0,0 +1,246 @@
+
+Take this example:
+```dart
+class MyNotifier extends StateNotifier<int> {
+  MyNotifier(this.period) : super(0) {
+    _timer = Timer.periodic(period, (t) => update());
+  }
+  final Duration period;
+  late final Timer _timer;
+
+  void update() => state++;
+
+  @override
+  void dispose() {
+    _timer.cancel();
+    super.dispose();
+  }
+}
+
+final myNotifierProvider = StateNotifierProvider<MyNotifier, int>((ref) {
+  final Duration period = ref.watch(durationProvider);
+  return MyNotifier(period);
+});
+```
+
+A first takeaway from this example is the following: *if `durationProvider` updates,
+`MyNotifier` disposes*.  
+As stated above, its instance is then re-created and its internal state re-initialized as usual.
+
+The second takeaway is that, unlike every other provider, the `dispose` callback is to be defined
+in the class, separately.  
+It is still possible to write `ref.onDispose` in its *provider*,
+but two different dispose callbacks are to be maintained.
+
+This API ambiguity is solved with `Riverpod 2.0`.  
+Check the above example, migrated to its `Notifier` equivalent.
+<AutoSnippet language="dart" {...oldLifecycles}></AutoSnippet>
+
+Here, instead, `MyNotifier`, will *not* dispose when `durationProvider` updates:
+only its *internal state* will.  
+This breaking change simplifies the API and, hopefully, your DX, for two reasons:
+1. there is only one lifecycle to take care of, the one we really care about: `Notifier`'s inner state;
+2. there is only one API to register a `dispose` action: `ref.onDispose`.
+
+### There is no `.mounted` property
+WIP
+
+### Final remarks
+When transitioning to `Notifier` (or `AsyncNotifier`), it is good to keep in mind some more differences.
+
+#### Migrating from `.addListener` and `.stream`
+`StateNotifier`'s `.addListener` and `.stream` can be used to listen for state changes.
+These two APIs aren't a thing within `Notifier` or `AsyncNotifier`.
+
+This is intentional due to the API uniformity `Notifier` and `AsyncNotifier` offer.
+Indeed, using a `Notifier` or an `AsyncNotifier` shouldn't be any different from any other provider.
+
+So, if you want to listen to a `Notifier`, just use `ref.listen`, [as explained here].
+
+#### Migrating from `.debugState`
+`StateNotifier` exposed a `.debugState` getter that isn't a thing on `Notifier`.  
+This property had some use within `state_notifier`, as it enabled accessing state
+from outside the class when in development mode, aka for testing purposes.
+
+Again, for uniformity reasons, it doesn't make sense to expose such property on a `Notifier`
+(or on an `AsyncNotifier`).  
+If you want to interact with `Notifier` or `AsyncNotifier` in a testing environment, you should do
+it just like with any other provider: by reading them directly (as opposed of directly instantiating
+a `Notifier` by hand).
+
+This [testing guide] fully explains how to use `ProviderContainer.read` to do so.
+
```

Comment:
The page needs formatting. There's inconsistent newlines. Some headings have a newline after them, some don't. And sometimes like here there's multiple newlines at once

You can install the MDX vscode extension to format those files

Reply:
I do have [this extension](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx) installed, but it doesn't offer any formatting options. Could you guide me into this, or could you show the correct extension to achieve the formatting you desire?

Reply:
Also I've tried to use Prettier, but I can't see any config files in `/website`. Let me know if the latest "format" commit is enough.

Reply:
@rrousselGit sorry, I have to ask you this one once again. I'm not sure how to solve this.